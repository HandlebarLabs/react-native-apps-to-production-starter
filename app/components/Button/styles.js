import { StyleSheet } from 'react-native';
import colors from '../../config/colors';

export default StyleSheet.create({
  container: {
    alignItems: 'center',
  },
  wrapper: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  icon: {
    width: 19,
    marginRight: 11,
  },
  text: {
    color: colors.$white,
    fontSize: 14,
    paddingVertical: 20,
    fontWeight: '300',
  },
});
