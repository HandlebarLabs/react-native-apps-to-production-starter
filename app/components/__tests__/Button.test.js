import React from 'react';
import renderer from 'react-test-renderer';
import 'react-dom';
import { TouchableOpacity } from 'react-native';
import { shallow } from 'enzyme';

import { ClearButton, styles } from '../Button';

it('renders successfully', () => {
  const rendered = renderer.create(<ClearButton />).toJSON();
  expect(rendered).toBeTruthy();
});

it('styles is an object', () => {
  expect(typeof styles).toBe('object');
});

it('renders custom text pass via props', () => {
  const rendered = renderer.create(<ClearButton text="hey this is a test" />).toJSON();
  expect(rendered).toMatchSnapshot();
});

it('handles a press', () => {
  const callback = jest.fn();
  const wrapper = shallow(<ClearButton text="test 1" onPress={callback} />);

  expect(wrapper.find(TouchableOpacity).length).toBe(1);
  wrapper
    .find(TouchableOpacity)
    .first()
    .props()
    .onPress();

  expect(callback.mock.calls.length).toBe(1);
});
