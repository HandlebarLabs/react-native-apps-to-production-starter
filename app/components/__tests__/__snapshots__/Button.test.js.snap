// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders custom text pass via props 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../app/components/Button/images/icon.png",
        }
      }
      style={
        Object {
          "marginRight": 11,
          "width": 19,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 14,
          "fontWeight": "300",
          "paddingVertical": 20,
        }
      }
    >
      hey this is a test
    </Text>
  </View>
</View>
`;
